src/SFA.DAS.ApplyService.InternalApi/Services/GatewayApiChecksService.cs

```diff
@@ -95,22 +115,40 @@ private async Task LookupCompaniesHouseDetails(ApplyGatewayDetails applyGatewayD
         private async Task LookupCharityCommissionDetails(ApplyGatewayDetails applyGatewayDetails, string charityNumberFromUkrlp)
         {
             if (!int.TryParse(charityNumberFromUkrlp, out var charityNumber))
-            { 
-                _logger.LogError("Unable to parse charity registration number from charity number in ukrlp `{charityNumber}`", charityNumberFromUkrlp);
-                return;
+            {
+                var message =
+                    $"Unable to parse charity registration number from charity number in ukrlp: '{charityNumberFromUkrlp}'";
+                _logger.LogError(message);
+                throw new ServiceUnavailableException(message);
```

Comment:
Where is this exception being handled?

Reply:
This is an service only used by Gateway, and gateway picks up the thrown exception and sends the user to a 'service unavailable' page when it happens (see the companion branch in das-roatp-gateway)